src/main/java/com/example/demo/controllers/rest/RoleController.java

```diff
@@ -0,0 +1,55 @@
+package com.example.demo.controllers.rest;
+
+import com.example.demo.db.entities.Role;
+import com.example.demo.db.entities.Users;
+import com.example.demo.db.repo.RoleRepo;
+import com.example.demo.db.repo.UserRepo;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Controller that implements the Role REST API
+ */
+@RestController
+@RequestMapping(path = "/roles")
+public class RoleController extends RestApiAbstract<Role, RoleRepo,Integer> {
+
+    final Logger logger = LogManager.getLogger(RoleController.class);
+    final RoleRepo repo;
+
+    public RoleController(RoleRepo repo) {
+        this.repo = repo;
+    }
+
+    @RequestMapping(path = "/{id}/role",method = RequestMethod.GET)
+    public Collection<String> getProfiles(@PathVariable Integer id){
```

Comment:
Please add here what this route does (JavaDoc) 
You can see the example in the RestApiAbstract.java